src/core/src/main/java/org/apache/jmeter/gui/action/template/Template.java

```diff
@@ -105,92 +104,44 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
         if (obj == null) {
             return false;
         }
         if (getClass() != obj.getClass()) {
             return false;
         }
         Template other = (Template) obj;
-        if (description == null) {
-            if (other.description != null) {
-                return false;
-            }
-        } else if (!description.equals(other.description)) {
+        if (!Objects.equals(description, other.description)) {
             return false;
         }
-        if (fileName == null) {
-            if (other.fileName != null) {
-                return false;
-            }
-        } else if (!fileName.equals(other.fileName)) {
+        if (!Objects.equals(fileName, other.fileName)) {
             return false;
         }
         if (isTestPlan != other.isTestPlan) {
             return false;
         }
-        if (name == null) {
-            if (other.name != null) {
-                return false;
-            }
-        } else if (!name.equals(other.name)) {
+        if (!Objects.equals(name, other.name)) {
             return false;
         }
-        if (!mapsEquals(parameters, other.parameters)) {
+        if (!Objects.equals(parameters, other.parameters)) {
             return false;
         }
         if (parent == null) {
-            if (other.parent != null) {
-                return false;
-            }
-        } else if (!parent.equals(other.parent)) {
-            return false;
+            return other.parent == null;
+        } else {
+            return parent.equals(other.parent);
```

Comment:
Sorry for nit-picking, but this `if-else` looks like `return Objects.equals(parent, other.parent)`

Reply:
then the whole thing can be collapsed to `return Object.equals(...) && Object.equals(...)  && Object.equals(...) ...`

Reply:
I've actually just re-generated `equals` and `hashCode`, much cleaner now :)